components/report/ReportModal.vue

```diff
@@ -0,0 +1,174 @@
+<script setup lang="ts">
+import type { mastodon } from 'masto'
+import { toggleBlockAccount, toggleFollowAccount, toggleMuteAccount, useRelationship } from '~~/composables/masto/relationship'
+
+const { account, status } = defineProps<{
+  account: mastodon.v1.Account
+  status: mastodon.v1.Status
+}>()
+
+const emit = defineEmits<{
+  (event: 'close'): void
+}>()
+
+const { client } = useMasto()
+
+const step = ref('choose')
+const serverRules = ref((await client.value.v2.instance.fetch()).rules || [])
+const reportReason = ref('')
+const selectedRuleIds = ref([])
+const additionalComments = ref('')
+const forwardReport = ref(false)
+
+const dismissButton = ref<HTMLDivElement>()
+
+async function submitReport() {
+  await client.value.v1.reports.create({
+    accountId: account.id,
+    statusIds: status?.id ? [status.id] : null,
```

Comment:
it would be nice to submit more statuses. Mastodon does it in its reporting view. How complex would it be to add this feature?

Reply:
Yes, this is an intentional decision I made but we should run it by Julia/product.
My reasoning:
- Most other social networks do not have a "select multiple statuses" feature. As someone new to Mastodon I found it confusing/unnecessary
- I don't think adding an extra "step" for selecting status would be difficult from a dev standpoint. I'm more concerned about the added complexity for the user. When Julia and I were walking through the Mastodon flow, we felt that there were a lot of steps which adds friction. I felt the statuses one was unnecessary / would not be used frequently, so I removed it, and then consolidated the other two steps into one page.

Reply:
Ok talked with Julia; I think I'll build it now. Easier to remove it later than add it in!

Reply:
done